include/cutlass/epilogue/fusion/xe_visitor.hpp

```diff
@@ -205,4 +205,230 @@ struct XeAuxLoad {
   }
 };
 
+template<
+  int Stages,
+  class CtaTileShapeMNK,
+  class ElementInput_,
+  class ElementCompute = cute::remove_pointer_t<ElementInput_>,
+  class StrideMNL_ = Stride<_0,_1,_0>,
+  int Alignment = 128 / sizeof_bits_v<cute::remove_pointer_t<ElementInput_>>,
+  bool EnableNullptr = true // Fallback scalar broadcast for nullptr params
+>
+struct XeRowBroadcast {
+  using StrideMNL = StrideMNL_;
+  // Get base element input type.
+  using ElementInput = cute::remove_pointer_t<ElementInput_>;
+  // Check if input is an array of pointers.
+  static constexpr bool IsArrayOfPointers = is_same_v<ElementInput*, ElementInput_>;
+  using PtrRowType = cute::conditional_t<IsArrayOfPointers, ElementInput const* const*, ElementInput const*>;
+
+  static_assert(Stages == 0, "Row broadcast doesn't support smem pipelining");
+
+  static constexpr bool IsDynamicBroadcast = is_same_v<remove_cvref_t<decltype(get<1>(StrideMNL{}))>, bool>; // row vector or scalar broadcast
+  static_assert(is_static_v<decltype(take<0,2>(StrideMNL{}))> || IsDynamicBroadcast); // batch stride can be dynamic or static
+  static_assert(take<0,2>(StrideMNL{}) == Stride<_0,_1>{} || IsDynamicBroadcast);
```

Comment:
Can you add error messages to the asserts?

Reply:
Yep, done cheers :+1: